lib/id_generator.rb

```diff
@@ -1,46 +1,22 @@
-require 'securerandom'
-
-class IdGenerator
-  VERSION = '0.1.2'.freeze
-
-  COUNTER_PART_SIZE = 8
-  CONTEXT_PART_SIZE = 2
-
-  RANDOM_PART_BYTES = 11
-  COUNTER_START = Time.new(2000).to_i
-
-  class Error < StandardError
-  end
-
-  def initialize(context_id)
-    raise(IdGenerator::Error, 'Invalid project id') unless context_id_valid?(context_id)
-
-    @context_id = value_to_hex(context_id, CONTEXT_PART_SIZE)
-  end
-
-  def generate
-    "#{time}-#{@context_id}-#{random_number}"
-  end
-
-  private
-
-  def context_id_valid?(context_id)
-    return false unless context_id.is_a?(Integer)
-    return false unless context_id.between?(0, 255)
-
-    true
+require 'id_generator/version'
+require 'id_generator/configuration'
+require 'id_generator/errors'
+require 'id_generator/generators/timestamped_random'
+
+module IdGenerator
+  def self.configure
+    yield(configuration)
```

Comment:
You can use here `Singleton` core ruby module, as a good alternative.

Reply:
Can you use Singleton for modules? In this case, gems are usually designed in a way that you have a top module with configuration that using instances defined inside of a gem